src/daemon.rs

```diff
@@ -33,21 +37,57 @@ fn parse_hash<T: Hash>(value: &Value) -> Result<T> {
     .chain_err(|| format!("non-hex value: {}", value))?)
 }
 
-fn header_from_value(value: Value) -> Result<BlockHeader> {
+fn header_from_value(value: Value, network: Network) -> Result<BlockHeader> {
     let header_hex = value
         .as_str()
         .chain_err(|| format!("non-string header: {}", value))?;
     let header_bytes = hex::decode(header_hex).chain_err(|| "non-hex header")?;
-    Ok(
-        deserialize(&header_bytes)
-            .chain_err(|| format!("failed to parse header {}", header_hex))?,
-    )
+    if network == Network::Dogecoin || network == Network::Dogetest {
+        let header_bytes = hex::decode(header_hex).chain_err(|| "non-hex header")?;
+        let mut cur = std::io::Cursor::new(&header_bytes);
+        let block_header: bitcoin::BlockHeader =
+            Decodable::consensus_decode(&mut cur).chain_err(|| "Unable to decode header")?;
+        Ok(block_header)
+    } else {
+        Ok(deserialize(&header_bytes)
+            .chain_err(|| format!("failed to parse header {}", header_hex))?)
+    }
 }
 
-fn block_from_value(value: Value) -> Result<Block> {
+fn block_from_value(value: Value, network: Network) -> Result<Block> {
     let block_hex = value.as_str().chain_err(|| "non-string block")?;
     let block_bytes = hex::decode(block_hex).chain_err(|| "non-hex block")?;
-    Ok(deserialize(&block_bytes).chain_err(|| format!("failed to parse block {}", block_hex))?)
+
+    let mut cur = std::io::Cursor::new(&block_bytes);
+    let block_header: bitcoin::BlockHeader =
+        Decodable::consensus_decode(&mut cur).chain_err(|| "Unable to decode header")?;
+
+    match network {
+        Network::Dogecoin | Network::Dogetest if block_header.version & 1 << 8 != 0 => {
+            let _: bitcoin::Transaction =
```

Comment:
1. Please extract this into a helper function
2. Use constants for magic numbers
3. Add documentation for when was this logic taken from, and where this ideally should be put in (rust-bitcoin)?